homeassistant/components/vallox/__init__.py

```diff
@@ -293,3 +321,39 @@ async def async_handle(self, call: ServiceCall) -> None:
         # be observed by all parties involved.
         if result:
             await self._coordinator.async_request_refresh()
+
+
+class ValloxEntity(CoordinatorEntity):
+    """Representation of a Vallox entity."""
+
+    coordinator: ValloxDataUpdateCoordinator
+    _device_config_url: str | None
+
+    def __init__(self, name: str, coordinator: ValloxDataUpdateCoordinator) -> None:
+        """Initialize a Vallox entity."""
+        super().__init__(coordinator)
+
+        if self.coordinator.config_entry is not None:
+            self._device_config_url = (
+                f"http://{self.coordinator.config_entry.data[CONF_HOST]}"
+            )
+        else:
+            self._device_config_url = None
+
+        self._device_uuid = self.coordinator.data.get_uuid()
+        self._device_model = self.coordinator.data.get_model()
+        self._device_sw_version = self.coordinator.data.get_sw_version()
+
+        self._attr_name = name
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device information of the entity."""
+        return DeviceInfo(
+            configuration_url=self._device_config_url,
+            identifiers={(DOMAIN, str(self._device_uuid))},
+            manufacturer=DEFAULT_NAME,
+            model=self._device_model,
+            name=self._attr_name,
```

Comment:
This is a bit interesting:

<img width="1427" alt="bild" src="https://user-images.githubusercontent.com/1417619/156237105-a6387d1f-297b-4832-ae95-9f901be7292b.png">

Why is the device called "Vallox Post Heater"? 🤔

Reply:
Hmm. I just re-tried with a completely clean setup, and there everything looks as expected:

![image](https://user-images.githubusercontent.com/4130005/156244153-baf15b12-de4c-44bd-a767-756108ccde8c.png)

I wonder what is different on your side. Did you try it on top of an existing configuration?

Reply:
Ah wait. The second time I started it, I can see this too now. I'll investigate.

Reply:
I fail to reliably reproduce this. I've seen it once so far, but haven't discovered a pattern yet...

Converting to draft again for the time being.

Reply:
It is somehow connected to changing a sensor name. Example:

1. Clean setup
2. Start HA, add Vallox integration. Device name should be fine.
3. Stop HA
4. In `binary_sensor.py`, add some new characters to `_attr_name`.
5. Start HA.
6. Observe the glitch.

Reply:
Okay, now it occured to me. `device_info` should never have used `_attr_name` for construction `DeviceInfo` in the first place, because this attribute obviously changes in the course of device init.

Seems like `device_info` sometimes was called before the topmost child class's `__init__` was running, fooling me that my code was correct.

Reply:
Ah, that makes sense 👍